mindmeld/components/question_answerer.py

```diff
@@ -372,13 +373,35 @@ class NativeQuestionAnswerer(BaseQuestionAnswerer):
     """
 
     # a common resource loaded generally used across all resolvers during loading process
-    RESOURCE_LOADER = ResourceLoader.create_resource_loader(
-        app_path=None,
-        text_preparation_pipeline=TextPreparationPipelineFactory.create_text_preparation_pipeline(
-            language="en",
-            tokenizer=WhiteSpaceTokenizer()
-        )
-    )
+    # a class var because it is also used in the underlying `Indices` class
+    RESOURCE_LOADER = None
+
+    def __init__(self, *args, **kwargs):
+        """
+        Args:
+            resource_loader (ResourceLoader, optional): A resource loader object used by the
+                underlying entity resolver models
+        """
+        super().__init__(*args, **kwargs)
+
+        # update class' resource loader; use one if already passed-in
+        resource_loader = kwargs.get("resource_loader")
+        if not resource_loader:
+            text_preparation_pipeline = \
+                TextPreparationPipelineFactory.create_text_preparation_pipeline(
+                    language="en",
+                    tokenizer=WhiteSpaceTokenizer()
+                )
+            query_factory = QueryFactory.create_query_factory(
+                app_path=None,
+                text_preparation_pipeline=text_preparation_pipeline,
+                duckling=True
+            )
+            resource_loader = ResourceLoader.create_resource_loader(
+                app_path=None,
+                query_factory=query_factory
+            )
+        NativeQuestionAnswerer.RESOURCE_LOADER = resource_loader
```

Comment:
Curious, why is this variable static instead of instance?

Reply:
Because it is also used in classes defined within NativeQuestionAnswerer-- Search and Indices. (for e.g. a resource loader is also required when loading dumped NativeQA's entity resolver objects. Since there are no .dump() and .load() methods for question-answerers, dumping and loading  is done implicitly when load_kb is called which in turn calls a method in Indices class. also, changing this to instance var also needs changes to execute() method of Search)